pytest_fixtures/core/contenthosts.py

```diff
@@ -41,93 +43,115 @@ def host_conf(request):
     return conf
 
 
+def host_post_config(hosts, config_name):
+    """A function that runs a specified post config on a list of content hosts."""
+    broker_args = settings.content_host.host_post_configs.get(config_name).to_dict()
+    for host in hosts:
+        for key, val in broker_args.items():
+            if "{" in val:
+                broker_args[key] = val.format(host=host)
+        Broker(**broker_args).execute()
```

Comment:
This is might be a random silly question, but it doesn't seem obvious on the surface to me upon an initial quick read.

My brain got caught up on the "why are we telling broker to take an action here, when it wasn't taking action before"

So in digging further, I'm inferring the following:

With the new approach, we now have actions we have to actually take time to execute.  In this case, either the 'enable-fips' workflow, or the 'enable-fapolicyd' workflow, as part of the 'factory<=>builder" relationship we're introducing.  When we encounter this, we then tell broker to take these specific actions before moving on to the next thing, which in this case would be yeilding the host back up to whatever called the contenthost_factory.

Does that seem generally correct?

Reply:
that's exactly right! it's like making sure the manufacturer puts all the extras on your vehicle before it leaves the factory.